### ISSUE TITLE
Open File Dialog Up Dir ("..") Stops Working

### ISSUE THREAD

**Issue by shoep** [microsoft/vscode#209848]
Type: <b>Bug</b>

I am working with files on a Unix host. Since the last update, when I choose File > Open File and use the ".." option to go up a directory, it works one or two levels up but beyond that it stops working. This always used to work. I can highlight the path and go that way, but the ".." button is much more efficient. Please take a look - thank you!

VS Code version: Code 1.88.0 (5c3e652f63e798a5ac2f31ffd0d863669328dc4c, 2024-04-03T13:26:18.741Z)
OS version: Windows_NT x64 10.0.19045
Modes:
Remote OS version: Linux x64 4.18.0-513.18.1.el8_9.x86_64

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10850H CPU @ 2.70GHz (12 x 2712)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.78GB (2.42GB free)|
|Process Argv|--crash-reporter-id 0f055dca-3865-4825-a464-4b21e1448316|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: structuredev13|
|OS|Linux x64 4.18.0-513.18.1.el8_9.x86_64|
|CPUs|Intel(R) Xeon(R) Gold 5317 CPU @ 3.00GHz (24 x 3600)|
|Memory (System)|250.84GB (223.77GB free)|
|VM|0%|
</details><details><summary>Extensions (23)</summary>

Extension|Author (truncated)|Version
---|---|---
remote-ssh|ms-|0.110.1
remote-ssh-edit|ms-|0.86.0
remote-explorer|ms-|0.4.3
emacs-mcx|tut|0.58.0
protein-viewer|Ari|0.1.0
codium|Cod|0.8.1
vscode-markdownlint|Dav|0.54.0
xml|Dot|2.5.1
copilot|Git|1.178.0
copilot-chat|Git|0.14.0
vscode-perl|hen|1.17.0
svn-scm|joh|2.17.0
vscode-js-annotations|lan|0.11.0
markdown-shortcuts|mdi|0.12.0
rainbow-csv|mec|3.11.0
cmake-tools|ms-|1.17.17
cpptools|ms-|1.19.9
cpptools-extension-pack|ms-|1.3.0
funclist|qrt|7.6.3
quicktype|qui|12.0.46
cmake|twx|0.0.17
sql|vsc|1.0.0
asn1|wil|0.5.3

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:30899288
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
h48ei257:31000450
pythontbext0:30879054
accentitlementst:30995554
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
8082a590:30971561
fegfb526:30981948
bg6jg535:30979843
ccp2r3:30993541
dsvsc020:30976470
pythonait:31006305
gee8j676:31009558
dsvsc021:30996838
gd77d436:30999572

```

</details>

**Comment by BenBeyond** [comment]
Same issue with the "Open Folder" dropdown when using the SSH container extension.  Just to add scope to this issue.

**Comment by chrmarti** [comment]
@TylerLeonhardt Seeing this with the simple Open Folder dialog that is using QuickPick when using the mouse to click `..` several times. Keyboard interaction shows similar issues. Maybe something with the events on the QuickPick changed?

**Comment by TylerLeonhardt** [comment]
Figured out the issue here... sent a PR but need to run it by someone to make sure I'm using our tree widget correctly.

**Comment by ivaylo-madjarov** [comment]
This issue is also described here:
https://github.com/microsoft/vscode-remote-release/issues/9797

In addition to the ".." option ceasing to work after one click, there is the additional issue that if you have two folders that are named the same thing, you can't click through the second one. For example, if you start in /home/user/, then go to /home/user/folder/, then try to click through to /home/user/folder/folder/, it also fails. I believe these issues are related as they appeared at the same time.

I am on VS Code 1.88.1.

**Comment by ivaylo-madjarov** [comment]
@TylerLeonhardt this is still not resolved as of 1.89.1, please reopen this issue.

**Comment by TylerLeonhardt** [comment]
@ivaylo-madjarov the fix is in 1.90. You can try it out using [VS Code Insiders](http://code.visualstudio.com/insiders/)
